terraform-modules/aws/opensearch/main.tf

```diff
@@ -0,0 +1,139 @@
+resource "aws_cloudwatch_log_group" "opensearch_slow_logs" {
+  name              = "opensearch-slow-logs"
+  retention_in_days = 14
+}
+
+resource "aws_cloudwatch_log_resource_policy" "opensearch_slow_logs_policy" {
+  policy_name = "opensearch-slow-logs-policy"
+  policy_document = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Effect" : "Allow",
+        "Principal" : {
+          "Service" : "es.amazonaws.com"
+        },
+        "Action" : [
+          "logs:PutLogEvents",
+          "logs:CreateLogStream",
+          "logs:PutLogEventsBatch"
+        ],
+        "Resource" : "arn:aws:logs:${var.aws_region}:${var.account_id}:log-group:${aws_cloudwatch_log_group.opensearch_slow_logs.name}:*"
+      }
+    ]
+  })
+}
+
+resource "aws_opensearch_domain" "this" {
+  domain_name    = var.domain_name
+  engine_version = "OpenSearch_2.5"
+
+  cluster_config {
+    instance_type          = "r4.large.search"
+    zone_awareness_enabled = true
+    instance_count         = var.instance_count
+  }
+
+  ebs_options {
+    ebs_enabled = true
+    volume_size = 10
+  }
+
+  encrypt_at_rest {
+    enabled = true
+  }
+
+  node_to_node_encryption {
+    enabled = true
+  }
+
+  domain_endpoint_options {
+    enforce_https       = var.enforce_https
+    tls_security_policy = "Policy-Min-TLS-1-2-2019-07"
+  }
+
+  # the dynamic block creates a vpc_options block with the specified security group and subnet IDs.
+  # If the variable vpc_enabled is set to false, the dynamic block is not created, 
+  # and the aws_opensearch_domain resource will not include a vpc_options block, 
+  # creating the OpenSearch domain publicly.
+  dynamic "vpc_options" {
+    for_each = var.vpc_enabled ? [1] : []
+    content {
+      security_group_ids = [aws_security_group.opensearch_sg.id]
```

Comment:
should allow the user to pass in an additional set of security group IDs if the user wants to let other things access this which wasnt created by this module.

The user input should be a `list`.  Then use the `concat()` function to combine the list: https://developer.hashicorp.com/terraform/language/functions/concat

Reply:
updated!!